staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -224,6 +224,11 @@ func (a *Allocator) Allocate(ctx context.Context, node *v1.Node) (finalResult []
 						}
 					}
 				}
+				// At least one device is required for 'All' allocation mode.
+				if len(requestData.allDevices) == 0 {
```

Comment:
Based on [John's latest comment](https://github.com/kubernetes/kubernetes/pull/129560/files#r1933989537) this should ensure all devices are available, or `adminAccess` is true.